### PR TITLE
Refactor GraphQL File Upload Implementation

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose

Fixes: [#2838](https://github.com/ballerina-platform/ballerina-standard-library/issues/2838)

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec

### CODE REVIEW

File: docs/proposals/file_upload.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 _Owners_: @ThisaruGuruge @DimuthuMadushan     
 _Reviewers_: @shafreenAnfar @ThisaruGuruge       
 _Created_: 2021/12/17   
-_Updated_: 2021/12/17     
+_Updated_: 2022/04/07
```
Comment (ThisaruGuruge) [inline docs/proposals/file_upload.md:6]:
```suggestion
_Updated_: 2022/04/07    
```

File: docs/spec/spec.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 _Owners_: @shafreenAnfar @DimuthuMadushan @ThisaruGuruge  
 _Reviewers_: @shafreenAnfar @DimuthuMadushan @ldclakmal  
 _Created_: 2022/01/06  
-_Updated_: 2022/04/05 
+_Updated_: 2022/04/07
```
Comment (ThisaruGuruge) [inline docs/spec/spec.md:6]:
```suggestion
_Updated_: 2022/04/07    
```

**Review by ThisaruGuruge — APPROVED** [review]

File: docs/proposals/file_upload.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
...
```
Comment (shafreenAnfar) [inline docs/proposals/file_upload.md:6]: We need the space here.

**Comment by shafreenAnfar** [comment]
LGTM! I hope there are no breaking changes here.

**Pull request merged**